src/session/content/chat_action_bar.rs

```diff
@@ -282,13 +286,218 @@ impl ChatActionBar {
         self.save_message_as_draft();
 
         let imp = self.imp();
+        let mut bindings = imp.bindings.borrow_mut();
+        while let Some(binding) = bindings.pop() {
+            binding.unwatch();
+        }
 
         if let Some(ref chat) = chat {
             self.load_draft_message(chat.draft_message());
 
             imp.chat_action_in_cooldown.set(false);
+
+            let chat_expression = gtk::ConstantExpression::new(chat).upcast();
+            let permissions_expression = gtk::PropertyExpression::new(
```

Comment:
Instead of using the classical way of creating property expression, you can "chain" them from the previous ones. In this case you can do:

```rust
let permissions_expression = chat_expression.chain_property::<Chat>("permissions").upcast();
```